backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1802,14 +1802,24 @@ func (r *ResourceManager) IsAuthorized(ctx context.Context, resourceAttributes *
 	// based on the namespace field in the request.
 	errlist := make([]error, 0)
 	userIdentity := ""
+	var userGroups []string
 	for _, auth := range r.authenticators {
 		identity, err := auth.GetUserIdentity(ctx)
-		if err == nil {
-			userIdentity = identity
 
-			break
+		if err != nil {
+			errlist = append(errlist, err)
+			continue
 		}
-		errlist = append(errlist, err)
+
+		groups, err := auth.GetUserGroups(ctx)
```

Comment:
Alright, should we log the possibles errors encounters in the following functions called by the GetUserGroups implementation `singleHeaderFromMetadata` (backend/src/apiserver/auth/util.go) ? Or simply ignore it and return an empty array ?

Reply:
If the group header doesn't exist, we do not consider it as an error and no need to log. However, if the group header exists but the value is not in the right format we expect, yes we should log the error. In any case, we should return the empty group because we can still use the person's identity to move forward with the authentication.